include/ginkgo/core/matrix/dense.hpp

```diff
@@ -244,6 +244,13 @@ class Dense
     std::unique_ptr<LinOp> row_permute(
         const Array<int64> *permutation_indices) const override;
 
+    /**
+     * Fill the dense matrix with a given value.
+     *
+     * @param value  the value to be filled
+     */
+    void fill(const ValueType value) { this->values_.fill(value); }
```

Comment:
This **ONLY** works if no padding is used (`stride == size[1]`), which we can't expect. If you want to keep this functionality, you need to add a test with a stride and add an additional kernel for each executor respecting the stride.

Reply:
good catch. i will fix this